test/test_client.py

```diff
@@ -1614,6 +1615,33 @@ def test_direct_connection(self):
         with self.assertRaises(ConfigurationError):
             MongoClient(['host1', 'host2'], directConnection=True)
 
+    def test_continuous_network_errors(self):
+        def server_description_count():
+            i = 0
+            for obj in gc.get_objects():
+                try:
+                    if isinstance(obj, ServerDescription):
+                        i += 1
+                except ReferenceError:
+                    pass
+            return i
+        gc.collect()
+        with client_knobs(min_heartbeat_interval=0.003):
+            client = MongoClient(
+                'invalid:27017',
+                heartbeatFrequencyMS=3,
+                serverSelectionTimeoutMS=100)
+            initial_count = server_description_count()
+            self.addCleanup(client.close)
+            with self.assertRaises(ServerSelectionTimeoutError):
+                client.test.test.find_one()
+            gc.collect()
+            final_count = server_description_count()
+            # If a bug like PYTHON-2433 is reintroduced then too many
+            # ServerDescriptions will be kept alive and this test will fail:
+            # AssertionError: 4 != 22 within 5 delta (18 difference)
+            self.assertAlmostEqual(initial_count, final_count, delta=5)
```

Comment:
Nice test. What is behind the choice of using `delta=5`? Is that number `5` liable to change?

Reply:
5 is arbitrary but it allows for some tolerance for race conditions, like having several clients open in the background (creating and destroying ServerDescriptions).